service/rest_patch.go

```diff
@@ -77,24 +77,35 @@ func (restapi restAPI) getPatchConfig(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if projCtx.Version == nil {
-		gimlet.WriteJSONInternalError(w, fmt.Sprintf("cannot get parser project for patch '%s' because version is nil", projCtx.Patch.Id.Hex()))
-		return
-	}
-
 	settings := restapi.GetSettings()
-	pp, err := model.ParserProjectFindOneByID(r.Context(), &settings, projCtx.Version.ProjectStorageMethod, projCtx.Version.Id)
-	if err != nil {
-		gimlet.WriteJSONInternalError(w, errors.Wrapf(err, "finding parser project '%s' for patch '%s'", projCtx.Version.Id, projCtx.Patch.Id.Hex()))
-		return
-	}
-	if pp == nil {
-		gimlet.WriteJSONInternalError(w, fmt.Sprintf("parser project '%s' for patch '%s' not found", projCtx.Version.Id, projCtx.Patch.Id.Hex()))
+	var pp *model.ParserProject
+	var err error
+	if projCtx.Patch.ProjectStorageMethod != "" {
+		pp, err = model.ParserProjectFindOneByID(r.Context(), &settings, projCtx.Version.ProjectStorageMethod, projCtx.Patch.Id.Hex())
+		if err != nil {
```

Comment:
nit: can we move the err and pp == nil check outside of the if so we dont duplicate it?

Reply:
I'd prefer to keep them separate so we can keep the error message wraps separate, which helps know which code path it took.